write-fonts/src/tables/cmap.rs

```diff
@@ -40,7 +40,12 @@ impl CmapSubtable {
 
         let mut prev = (u16::MAX - 1, u16::MAX - 1);
         for (cp, gid) in mappings {
-            let gid = gid.to_u16();
+            let gid = gid.to_u32();
+            if gid > 0xFFFF {
+                // Should we just fail here?
+                continue;
+            }
+            let gid = gid as u16;
```

Comment:
(generating a cmap4 subtable)

I'm not sure what to do here. I assume if we're generating a font with more than 64k glyphs that we just shouldn't emit a cmap4 subtable.